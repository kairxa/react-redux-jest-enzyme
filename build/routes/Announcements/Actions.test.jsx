/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Actions from './Actions';

test('should render equal to snapshot', () => {
  const mock = jest.fn();
  const container = renderer.create(
    <Actions
      onChangeSearchString={mock}
      onChangeSearchDate={mock}
      onSubmitSearchString={mock}
      onSubmitSearchDate={mock}
    />
  );
  const tree = container.toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing isRequired props
test('should throw error when one or more function is not provided', () => {
  const mock = jest.fn();
  let container = renderer.create(<Actions />);
  let tree = container.toJSON();
  expect(tree).toThrowErrorMatchingSnapshot();

  container = renderer.create(
    <Actions
      onChangeSearchDate={mock}
      onSubmitSearchString={mock}
      onSubmitSearchDate={mock}
    />
  );
  tree = container.toJSON();
  expect(tree).toThrowErrorMatchingSnapshot();

  container = renderer.create(
    <Actions
      onChangeSearchString={mock}
      onSubmitSearchString={mock}
      onSubmitSearchDate={mock}
    />
  );
  tree = container.toJSON();
  expect(tree).toThrowErrorMatchingSnapshot();

  container = renderer.create(
    <Actions
      onChangeSearchString={mock}
      onChangeSearchDate={mock}
      onSubmitSearchDate={mock}
    />
  );
  tree = container.toJSON();
  expect(tree).toThrowErrorMatchingSnapshot();

  container = renderer.create(
    <Actions
      onChangeSearchString={mock}
      onChangeSearchDate={mock}
      onSubmitSearchString={mock}
    />
  );
  tree = container.toJSON();
  expect(tree).toThrowErrorMatchingSnapshot();
});

// Testing #search change event
test('should trigger onChangeSearchString when #search is changed', () => {
  const mock = jest.fn();
  const wrapper = shallow(
    <Actions
      onChangeSearchString={mock}
      onChangeSearchDate={mock}
      onSubmitSearchString={mock}
      onSubmitSearchDate={mock}
    />
  );

  wrapper.find('#search').simulate('change', 'string');
  expect(mock).toHaveBeenLastCalledWith('string');
  wrapper.find('#date').simulate('change', 'date');
  expect(mock).toHaveBeenLastCalledWith('date');
  wrapper.find('#searchAnnouncementsString').simulate('submit', 'searchString');
  expect(mock).toHaveBeenLastCalledWith('searchString');
  wrapper.find('#searchAnnouncementsDate').simulate('submit', 'searchDate');
  expect(mock).toHaveBeenLastCalledWith('searchDate');

  expect(mock.mock.calls.length).toBe(4);
});
