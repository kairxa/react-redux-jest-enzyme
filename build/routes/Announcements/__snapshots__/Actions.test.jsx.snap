exports[`test should render equal to snapshot 1`] = `
<section
  className="containerActions">
  <div
    className="containerActionsLeft">
    <form
      className="containerInput"
      id="searchAnnouncementsString"
      onSubmit={[Function]}>
      <input
        className="inputText"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="text" />
      <button
        className="inputAction">
        <test-file />
      </button>
    </form>
    <form
      className="containerInput"
      id="searchAnnouncementsDate"
      onSubmit={[Function]}>
      <input
        className="inputText"
        id="date"
        name="date"
        onChange={[Function]}
        placeholder="Search by Date (YYYY/MM/DD)"
        type="date" />
      <button
        className="inputAction"
        type="submit">
        <test-file />
      </button>
    </form>
  </div>
  <div
    className="containerActionsRight">
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}>
      NEW ANNOUNCEMENT
    </a>
  </div>
</section>
`;

exports[`test should throw error when one or more function is not provided 1`] = `"received is not a function"`;

exports[`test should throw error when one or more function is not provided 2`] = `"received is not a function"`;

exports[`test should throw error when one or more function is not provided 3`] = `"received is not a function"`;

exports[`test should throw error when one or more function is not provided 4`] = `"received is not a function"`;

exports[`test should throw error when one or more function is not provided 5`] = `"received is not a function"`;
